gnomad/utils/vep.py

```diff
@@ -85,53 +92,118 @@
+
+    .. warning::
+        If no `vep_config_path` is supplied, this function will only work for Dataproc clusters
+        created with `hailctl dataproc start --vep`. It assumes that the command is `/path/to/vep`.
+
+    :param vep_config_path: Optional path to use as the VEP config file. If None, `VEP_CONFIG_URI` environment variable is used
+    :return: VEP help string
+    """
+    if vep_config_path is None:
+        vep_config_path = os.environ["VEP_CONFIG_URI"]
+
+    with hl.hadoop_open(vep_config_path) as vep_config_file:
+        vep_config = json.load(vep_config_file)
+        vep_command = vep_config["command"]
+        vep_help = subprocess.check_output([vep_command[0]]).decode("utf-8")
+        return vep_help
+
+
+def get_vep_context(ref: Optional[str] = None) -> VersionedTableResource:
+    """
+    Get VEP context resource for the genome build `ref`
+
+    :param ref: Genome build. If None, `hl.default_reference` is used
+    :return: VEPed context resource
+    """
+    import gnomad.resources.grch37.reference_data as grch37
+    import gnomad.resources.grch38.reference_data as grch38
 
+    if ref is None:
+        ref = hl.default_reference().name
 
-def vep_config_path(ref: str = "GRCh37"):
-    if ref not in VEP_REFERENCE_DATA.keys():
-        raise DataException(
-            "Select reference as one of: {}".format(",".join(VEP_REFERENCE_DATA.keys()))
+    if ref not in POSSIBLE_REFS:
+        raise ValueError(
+            f'get_vep_context passed {ref}. Expected one of {", ".join(POSSIBLE_REFS)}'
         )
-    return VEP_REFERENCE_DATA[ref]["vep_config"]
 
+    vep_context = grch37.vep_context if ref == "GRCh37" else grch38.vep_context
+    return vep_context
 
-def vep_or_lookup_vep(ht, reference_vep_ht=None, reference=None, vep_config=None):
+
+def vep_or_lookup_vep(
+    ht, reference_vep_ht=None, reference=None, vep_config_path=None, vep_version=None
+):
     """
     VEP a table, or lookup variants in a reference database
 
+    .. warning::
+        If `reference_vep_ht` is supplied, no check is performed to confirm `reference_vep_ht` was
+        generated with the same version of VEP / VEP configuration as the VEP referenced in `vep_config_path`.
+
     :param ht: Input Table
     :param reference_vep_ht: A reference database with VEP annotations (must be in top-level `vep`)
     :param reference: If reference_vep_ht is not specified, find a suitable one in reference (if None, grabs from hl.default_reference)
-    :param vep_config: vep_config to pass to hl.vep (if None, a suitable one for `reference` is chosen)
-    :return: VEPped Table
+    :param vep_config_path: vep_config to pass to hl.vep (if None, a suitable one for `reference` is chosen)
+    :param vep_version: Version of VEPed context Table to use (if None, the default `vep_context` resource will be used)
+    :return: VEPed Table
     """
+    vep_help = get_vep_help(vep_config_path)
```

Comment:
If the `vep_config_path` argument is None, this will default to the value of `os.environ["VEP_CONFIG_URI"]` whereas `vep_config_path` will default the constant `VEP_CONFIG_PATH`. To avoid the inconsistency, this could be moved below:
```python
if vep_config_path is None:
    vep_config_path = VEP_CONFIG_PATH
```

Reply:
Thank you!